awx/main/tests/functional/api/test_instance.py

```diff
@@ -54,3 +54,27 @@ def test_health_check_usage(get, post, admin_user):
     get(url=url, user=admin_user, expect=200)
     r = post(url=url, user=admin_user, expect=200)
     assert r.data['msg'] == f"Health check is running for {instance.hostname}."
+
+
+def test_hostnames(post, admin_user):
+    url = reverse('api:instance_list')
+    for value in ["foobar", "foo.bar.baz"]:
+        post(url=url, user=admin_user, data={"hostname": value}, expect=201)
+    for value in ["--yoooo", "$3$@foobar@#($!@#*$"]:
+        post(url=url, user=admin_user, data={"hostname": value}, expect=400)
+    for value in ["localhost", "127.0.0.1"]:
+        post(url=url, user=admin_user, data={"hostname": value}, expect=400)
```

Comment:
I think these tests just need a `mocker.patch` to make it believe it's an OCP deployment... or maybe a use of `override_settings`

Reply:
Yeah, mock is good to patch a simple method, and less good at patching data. Settings is more data than methods, balanced on top of a gazillion wrappers, hacks, and customizations.